packages/playground/remote/src/lib/worker-thread.ts

```diff
@@ -186,6 +186,96 @@ export class PlaygroundWorkerEndpoint extends PHPWorker {
 	async replayFSJournal(events: FilesystemOperation[]) {
 		return replayFSJournal(this.__internal_getPHP()!, events);
 	}
+
+	async backfillStaticFilesRemovedFromMinifiedBuild() {
+		await backfillStaticFilesRemovedFromMinifiedBuild(
+			this.__internal_getPHP()!
+		);
+	}
+}
+
+async function backfillStaticFilesRemovedFromMinifiedBuild(php: PHP) {
+	if (!php.requestHandler) {
+		logger.warn('No PHP request handler available');
+		return;
+	}
+
+	try {
+		const remoteAssetListPath = joinPaths(
+			php.requestHandler.documentRoot,
+			'wordpress-remote-asset-paths'
+		);
+
+		/**
+		 * Don't download static assets if they're already downloaded.
+		 * WordPress may be loaded either from a production release or a minified bundle.
+		 * Minified bundles are shipped without most CSS files, JS files, and other static assets.
+		 * Instead, they contain a list of remote assets in wordpress-remote-asset-paths.
+		 * We use this list to determine if we should fetch them on demand or if they are already downloaded.
+		 * If the list is empty, we assume the assets are already downloaded.
+		 * See https://github.com/WordPress/wordpress-playground/pull/1531
+		 */
+		if (
+			!php.fileExists(remoteAssetListPath) ||
+			(await php.readFileAsText(remoteAssetListPath)) === ''
+		) {
+			return;
+		}
+		const wpVersion = await getLoadedWordPressVersion(php.requestHandler);
+		const staticAssetsDirectory =
+			wpVersionToStaticAssetsDirectory(wpVersion);
+		if (!staticAssetsDirectory) {
+			return;
+		}
+		const response = await fetch(
+			joinPaths('/', staticAssetsDirectory, 'wordpress-static.zip')
+		);
+
+		if (!response.ok) {
+			throw new Error(
+				`Failed to fetch WordPress static assets: ${response.status} ${response.statusText}`
+			);
+		}
+
+		const zipPath = '/tmp/wordpress-static-assets.zip';
+		await php.writeFile(
```

Comment:
The `unzipFile` helper will accept the File object directly

Reply:
Done [6f48337](https://github.com/WordPress/wordpress-playground/pull/1532/commits/6f48337ead916ee6456c37d1e0185f11f84680c8)